signer/tests/integration/transaction_coordinator.rs

```diff
@@ -5945,34 +5986,41 @@ async fn should_handle_dkg_coordination_failure() {
         is_epoch3: true,
     };
 
-    // We're verifying that the coordinator is currently
-    // processing requests correctly. Since we previously checked
-    // that 'should_coordinate_dkg' will trigger and we set the
-    // 'dkg_max_duration' to 10 milliseconds we expect that
-    // DKG will run & fail
-    let result = coordinator.process_new_blocks().await;
-    assert!(
-        result.is_ok(),
-        "process_new_blocks should complete successfully even with DKG failure"
-    );
+    // Coordinator should be the current signer, since the bootstrap
+    // signing set has only the current signer in it.
+    assert!(coordinator.is_coordinator(&chain_tip.block_hash));
+
+    // We're verifying that the coordinator is currently processing
+    // requests correctly. Since we previously checked that
+    // 'should_coordinate_dkg' will trigger and we set the
+    // 'dkg_max_duration' to 10 milliseconds we expect that DKG will run
+    // and fail.
+    coordinator
+        .process_new_blocks(chain_tip)
+        .await
+        .expect("process_new_blocks should complete successfully even with DKG failure");
 
     // Here we check that DKG ran & correctly failed by fetching
     // the latest DKG shares from storage. We test it failed
     // by asserting that the latest row is_none()
     let dkg_shares = storage.get_latest_encrypted_dkg_shares().await.unwrap();
     assert!(
         dkg_shares.is_none(),
-        "DKG shares should not exist since DKG failed to complete due to timeout"
+        "DKG shares should not exist since DKG failed to complete"
     );
 
-    // Verify that we can still process blocks after DKG failure,
-    // this final assert specifically checks that blocks are still
-    // being processed since there was an aggregate key to fallback on
-    let result = coordinator.process_new_blocks().await;
-    assert!(
-        result.is_ok(),
-        "Should be able to continue processing blocks after DKG failure"
-    );
+    // Verify that we can still process blocks after DKG failure, this
+    // final assert specifically checks that transaction processing still
+    // continues since there is signer set info to fallback on. Note that
+    // the coordinator should be emitting lots of error logs when
+    // constructing bitcoin or stacks transactions, but the coordinator
+    // should continue through all of that.
+    coordinator
+        .process_new_blocks(chain_tip)
+        .await
+        .expect("process_new_blocks should complete successfully even with DKG failure");
+
+    testing::storage::drop_db(db).await;
```

Comment:
This seems doing the same check as the previous `process_new_blocks`, how does that differ exactly? The fallback that is mentioned shouldn't apply also to the previous iteration?

Reply:
I don't think that it differs at all either. I suspect, as it was before this change, was supposed to change something and then call `process_new_blocks` again but I'm not sure what that was supposed to be. I'll just remove it.

Reply:
I removed it in https://github.com/stacks-sbtc/sbtc/pull/1793/commits/cbc450ee3b96693c64e5995d5f46b9806f6a971c